CONTRIBUTING.md

```diff
@@ -1,80 +1,233 @@
+be *rejected*.
+
+
+## RFC Contribution Stages
+
+RFCs are guided by a *champion* through a series of stages: *strawman*,
+*proposal*, *draft*, and *accepted* (or *rejected*), each of which has suggested
+entrance criteria and next steps detailed below. RFCs typically advance one
+stage at a time, however may advance multiple stages at a time. Stage
+advancements typically occur during
+[Working Group](https://github.com/graphql/graphql-wg) meetings, however may
+occur online.
+
+All RFCs start as either a *strawman* or *proposal*.
+
+## Stage 0: *Strawman*
+
+The *strawman* stage captures described problems or partially-considered
+solutions which lack the entrance criteria to be considered a *proposal*. A
+*strawman* may be an issue or a pull request (though an illustrative pull
+request is preferrable).
+
+*There is no entrance criteria for a Strawman*
+
+As implied by the name [strawman](https://en.wikipedia.org/wiki/Straw_man_proposal),
+the goal at this stage is to knock it down (*reject*) by considering other
+possible related solutions, showing that the motivating problem can be solved
+with no change to the specification, or that it is not aligned with the
+*guiding principles*.
+
+Once determined that the *strawman* is compelling, it should seek the entrance
+criteria for *proposal*.
+
+
+## Stage 1: *Proposal*
+
+An RFC at the *proposal* stage is a solution to a problem with enough fidelity
+to be discussed in detail, backed by a willing *champion*. A *proposal*'s goal
+is to make a compelling case for acceptance by describing both the problem and
+the solution via examples and spec edits. A *proposal* should be a pull request.
+
+*Entrance criteria:*
+
+* Identified *champion*
+* Clear explanation of problem and solution
+* Illustrative examples
+* Incomplete spec edits
+* Identification of potential concerns, challenges, and drawbacks
+
+A *proposal* is subject to the same discussion as a *strawman*: ensuring that it
+is well aligned with the *guiding principles*, is a problem worth solving, and
+is the preferred solution to that problem. A *champion* is not expected to have
+confidence in every detail at this stage and should instead focus on identifying
+and resolving issues and edge-cases. To better understand the ramifications of a
+*proposal*, a *champion* is encouraged to implement it in a GraphQL library.
+
+Most *proposals* are expected to evolve or change and may be rejected. Therefore,
+it is unwise to rely on a *proposal* in a production GraphQL service. GraphQL
+libraries *may* implement *proposals*, though are encouraged to not enable the
+*proposal* feature without explicit opt-in.
+
+
+## Stage 2: *Draft*
+
+An RFC at the *draft* stage is a fully formed solution which has working group
+consensus that it is a problem worth solving and the solution is preferred.
+A *draft's* goal is to precisely and completely describe the solution and
+resolve any implementation concerns. A *draft* must be a pull request.
```

Comment:
Thesis: fully formed solution.
Requirements:  working group "worth trying to solve" consensus along with pull request.
Why:  Goal is to precisely and completely describe the solution. Given a draft, different people must be able to implement identical behavior across GraphQL libraries.

```
An RFC at the *draft* stage is a fully formed solution. There is working
group consensus that the problem identified is worth solving, and this
particular solution is preferred. A *draft's* goal is to precisely and 
completely describe the solution and resolve any concerns through library
implementations. A *draft* must be a pull  request.
```